### PR TITLE
hub: payment attempts endpoint

### PR DESCRIPTION
**Pull request by jurgenwerk**
Ticket: CS-4664

We'll use this endpoint to fetch payment attempts (past payments). 

image

TODO: 
- [x] Filter by status

### CODE REVIEW

File: packages/hub/services/serializers/scheduled-payment-attempt-serializer.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    as: 'scheduledPaymentSerializer',
+  });
+
+  serialize(
+    model: ScheduledPaymentAttemptWithScheduledPayment | ScheduledPaymentAttemptWithScheduledPayment[]
+  ): JSONAPIDocument {
+    if (Array.isArray(model)) {
+      return {
+        data: model.map((model) => this.serialize(model).data),
+        included: uniqBy(model, 'scheduledPayment.id').map((model) => {
```
Comment (jurgenwerk) [inline packages/hub/services/serializers/scheduled-payment-attempt-serializer.ts:18]: Include scheduled payment as a related model in JSON:API fashion - https://jsonapi.org/format/#document-compound-documents

File: packages/hub/routes/scheduled-payment-attempts.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import ScheduledPaymentSerializer from '../services/serializers/scheduled-payment-serializer';
+import { ScheduledPayment, ScheduledPaymentAttempt, ScheduledPaymentAttemptStatusEnum } from '@prisma/client';
+import { isValidDate } from '../utils/dates';
+
+export type ScheduledPaymentAttemptWithScheduledPayment = ScheduledPaymentAttempt & {
+  scheduledPayment: ScheduledPayment;
+};
+
+export default class ScheduledPaymentAttemptsRoute {
+  scheduledPaymentValidator: ScheduledPaymentValidator = inject('scheduled-payment-validator', {
```
Comment (FadhlanR) [inline packages/hub/routes/scheduled-payment-attempts.ts:15]: We can remove this property

File: packages/hub/routes/scheduled-payment-attempts.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    as: 'scheduledPaymentValidator',
+  });
+  scheduledPaymentSerializer: ScheduledPaymentSerializer = inject('scheduled-payment-serializer', {
+    as: 'scheduledPaymentSerializer',
+  });
+  web3 = inject('web3-http', { as: 'web3' });
```
Comment (FadhlanR) [inline packages/hub/routes/scheduled-payment-attempts.ts:21]: This one also

File: packages/hub/routes/scheduled-payment-attempts.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    let status: ScheduledPaymentAttemptStatusEnum | undefined = undefined;
+    let statusQuery = ctx.query['filter[status]'] as string;
+    if (['failed', 'succeeded', 'inProgress'].includes(statusQuery)) {
+      status = statusQuery as ScheduledPaymentAttemptStatusEnum;
+    }
+
+    let prisma = await this.prismaManager.getClient();
+
+    let attempts = (await prisma.scheduledPaymentAttempt.findMany({
```
Comment (FadhlanR) [inline packages/hub/routes/scheduled-payment-attempts.ts:53]: Do we need pagination?

File: packages/hub/routes/scheduled-payment-attempts.ts (hunk 2 above)
Comment (jurgenwerk) [reply packages/hub/routes/scheduled-payment-attempts.ts]: Good catch. Removed.

**Comment by jurgenwerk** [comment]
@FadhlanR I think you asked if we need pagination but your question disappeared (probably because I rebased). My opinion is that we will eventually need it but not very important for now until we get the MVP shipped. It would be a good idea to come up with a general pagination solution that we can reuse in many endpoints. I'll add a ticket for that.

**Review by lukemelia — APPROVED** [review]
Nice!

**Pull request merged**